builtin/credential/approle/validation.go

```diff
@@ -335,23 +334,98 @@ func (b *backend) secretIDAccessorLock(secretIDAccessor string) *sync.RWMutex {
 	return lock
 }
 
+// nonLockedSecretIDStorageEntry fetches the secret ID properties from physical
+// storage. The entry will be indexed based on the given HMACs of both role
+// name and the secret ID. This method will not acquire secret ID lock to fetch
+// the storage entry. Locks need to be acquired before calling this method.
+func (b *backend) nonLockedSecretIDStorageEntry(s logical.Storage, roleNameHMAC, secretIDHMAC string) (*secretIDStorageEntry, error) {
+	if secretIDHMAC == "" {
+		return nil, fmt.Errorf("missing secret ID HMAC")
+	}
+
+	if roleNameHMAC == "" {
+		return nil, fmt.Errorf("missing role name HMAC")
+	}
+
+	// Prepare the storage index at which the secret ID will be stored
+	entryIndex := fmt.Sprintf("secret_id/%s/%s", roleNameHMAC, secretIDHMAC)
+
+	entry, err := s.Get(entryIndex)
+	if err != nil {
+		return nil, err
+	}
+	if entry == nil {
+		return nil, nil
+	}
+
+	result := secretIDStorageEntry{}
+	if err := entry.DecodeJSON(&result); err != nil {
+		return nil, err
+	}
+
+	persistNeeded := false
+	if result.SecretIDNumUsesDeprecated != 0 {
```

Comment:
What you suggest makes sense to me. Deprecation notice does not mean that the values can't be relied on right away. It should be valid as long as it is returned. However we chose to reset the  values for deprecated versions of `CreationTime`, `ExplicitMaxTTL`, et al in token store, but that was a different use case.

It probably may be a good idea to have an issue for every release version where in we list down all the things to be removed in that version.

Pushed up the suggestions.